test_env.py

```diff
@@ -95,3 +95,4 @@ class TestClassSimulator:
     def test_deployment(self, query_file, query_udfs):
         result = check_filter(query_file, query_udfs)
         assert result == util.EXIT_SUCCESS
+        #TODO: check the results in result.txt as well
```

Comment:
Ideally, we would write dedicate rust or python tests instead of relying on results.txt. We should have an API to query results.

Reply:
Do you mean use another file that is used inside of test_env, like check_filter?  Also, as far as an API goes, how would that work, given that the simulator process will be dead by the time we want to look at results?

Reply:
Write tests that use the simulator API directly. 
```bash
    for tick in 0..20 {
        simulator.tick(tick);
    }
```
In this loop we would query results from storage and assert if they are correct.

Reply:
Okay, thanks for the info.